### PR TITLE
feat: FIPS 2204 definitions

### PR DESCRIPTION
**Pull request by artunduman**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/sig_config.go [hunk 1]
```diff
@@ -477,6 +482,20 @@ var (
 		Version:       LinuxSIGImageVersion,
 	}
 
+	SIGUbuntuFipsContainerd2204ImageConfigTemplate = SigImageConfigTemplate{
+		ResourceGroup: AKSUbuntuResourceGroup,
+		Gallery:       AKSUbuntuGalleryName,
+		Definition:    "2204fipscontainerd",
+		Version:       "202404.09.0", // TODO(artunduman): Update version when the image is ready
```
Comment (artunduman) [inline pkg/agent/datamodel/sig_config.go:489]: Points to the image version being released [here](https://msazure.visualstudio.com/CloudNativeCompute/_releaseProgress?_a=release-pipeline-progress&releaseId=159123)

**Review by chengliangli0918 — APPROVED** [review]

File: pkg/agent/datamodel/sig_config.go (hunk 1 above)
Comment (cameronmeissner) [reply pkg/agent/datamodel/sig_config.go:489]: we have yet to add RP selection logic for these new SKUs right?

Comment (artunduman) [reply pkg/agent/datamodel/sig_config.go:489]: correct

Comment (cameronmeissner) [reply pkg/agent/datamodel/sig_config.go:489]: confirmed offline this has not been added yet

**Review by cameronmeissner — APPROVED** [review]

**Review by anujmaheshwari1 — APPROVED** [review]

**Pull request merged**